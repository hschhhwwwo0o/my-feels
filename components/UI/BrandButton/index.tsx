import React, { FunctionComponent, ReactNode } from "react";

interface IBrandButton {
  onClickHandler?: () => any;
  children?: ReactNode;
}

const BrandButton: FunctionComponent<IBrandButton> = ({ children, onClickHandler = () => {} }) => {
  return (
    <>
      <div>
        <button
          onClick={onClickHandler}
          className="w-full bg-[#151515] text-white py-[21px] rounded-md text-base px-5 font-semibold border-2 border-[#EAEAEA] dark:border-[#151515] transition-all duration-1000"
        >
          {children}
        </button>
      </div>
    </>
  );
};

export default BrandButton;
