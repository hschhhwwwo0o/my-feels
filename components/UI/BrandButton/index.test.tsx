import React from "react";
import renderer from "react-test-renderer";
import BrandButton from "./index";

test("Use BrandButton", () => {
  const component = renderer.create(<BrandButton>Oh click me, dear</BrandButton>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
