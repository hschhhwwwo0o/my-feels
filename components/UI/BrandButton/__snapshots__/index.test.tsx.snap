// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use BrandButton 1`] = `
<div>
  <button
    className="w-full bg-[#151515] text-white py-[21px] rounded-md text-base px-5 font-semibold border-2 border-[#EAEAEA] dark:border-[#151515] transition-all duration-1000"
    onClick={[Function]}
  >
    Oh click me, dear
  </button>
</div>
`;
